jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/LocalRegistry.java

```diff
@@ -17,77 +17,145 @@
 package com.google.cloud.tools.jib.registry;
 
 import com.google.cloud.tools.jib.Command;
-import com.google.common.io.CharStreams;
 import java.io.IOException;
-import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.UUID;
+import javax.annotation.Nullable;
 import org.junit.rules.ExternalResource;
-import org.junit.rules.TestRule;
 
-/** {@link TestRule} that runs a local registry. */
+/** Runs a local registry. */
 public class LocalRegistry extends ExternalResource {
 
+  private final String containerName = "registry-" + UUID.randomUUID();
   private final int port;
+  private final boolean pullBusyBox;
+  @Nullable private final String username;
+  @Nullable private final String password;
 
-  /** The name for the container running the registry. */
-  private final String containerName = "registry-" + UUID.randomUUID();
+  public LocalRegistry(int port, boolean pullBusyBox) {
```

Comment:
Maybe `pullBusybox` can be removed and have it be a method instead? (so `Blob...IntegrationTest` will call that method)